pytorch_lightning/loops/utilities.py

```diff
@@ -100,6 +97,9 @@ def _process_training_step_output(
     # map to results under the hood
     results.minimize = loss
 
+    if trainer.terminate_on_nan:
+        check_finite_loss(loss)
+
```

Comment:
reviewer: the check for the loss gets moved here, which happens when we get loss and results from the `LightningModule.training_step`